### PR TITLE
Tarball script version for master & 34

### PR DESCRIPTION
**Pull request by morlovich**

### CODE REVIEW

File: devel/create_distro_tarball.sh
Comment (hillsp) [inline devel/create_distro_tarball.sh:144]: wdyt about just having one variable with all this crap in, instead of multiple? Obviously there are trade-offs there

File: devel/create_distro_tarball.sh
Comment (hillsp) [inline devel/create_distro_tarball.sh]: I think maybe I'd say '--checkout commit_ish' which is the git way of saying "something that can be checked out.

File: devel/create_distro_tarball.sh
Comment (hillsp) [inline devel/create_distro_tarball.sh:64]: This should include --depth=. otherwise it wont work if your checkout isn't in a direcotry named 'src'. Apparently I missed that when I was messing with the build system.

File: devel/create_distro_tarball.sh
Comment (hillsp) [inline devel/create_distro_tarball.sh:73]: This would be SO MUCH more readable if it used getopt. Take a look here:

https://github.com/pagespeed/mod_pagespeed/blob/d55c2d7a2c64fab183f77571a18dad80ffe3cb1b/install/build_on_vm.sh#L34

File: devel/create_distro_tarball.sh
Comment (hillsp) [inline devel/create_distro_tarball.sh]: Beware of this; It turns out that on CentOS 6, --branch <tag> will print a warning but silently checkout master.

File: devel/create_distro_tarball.sh
Comment (morlovich) [reply devel/create_distro_tarball.sh:73]: Sure, but I feel like it's outside the scope of this.

File: devel/create_distro_tarball.sh
Comment (morlovich) [reply devel/create_distro_tarball.sh:144]: I've strongly considered it, the reason I picked this was to make it easier to see the two sides next to each other.

File: devel/create_distro_tarball.sh
Comment (morlovich) [reply devel/create_distro_tarball.sh]: I would find that less helpful.

File: devel/create_distro_tarball.sh
Comment (morlovich) [reply devel/create_distro_tarball.sh:64]: Could you elaborate? A lot of stuff assumes src/ below.

File: devel/create_distro_tarball.sh
Comment (hillsp) [reply devel/create_distro_tarball.sh]: OK, but what you have is objectively wrong since 34 isn't a tag.

Comment (morlovich) [reply devel/create_distro_tarball.sh]: Fair enough. Done, as suggested. (Not pushed yet, waiting for further feedback)

File: devel/create_distro_tarball.sh
Comment (hillsp) [reply devel/create_distro_tarball.sh:64]: Sorry, I wasn't reading carefully enough. It's fine as-is.

The background is that gyp looks for a directory containing "src" in the path to figure out the root (which it calls "depth", for some reason). If you build from a directory that isn't src, it doesn't work. The pattern we use everywhere else is:

git clone <some_dir>
cd <some_dir>
gyp --depth=.

That way we can run gyp in the root of the checkout and it doesn't matter what the checkout is called. Doing the same thing wouldn't be awful for consistency, but isn't required.

File: devel/create_distro_tarball.sh
Comment (hillsp) [reply devel/create_distro_tarball.sh:73]: How about the fact that I'm struggling to review the code because I don't understand how the arg parsing works?

File: devel/create_distro_tarball.sh
Comment (hillsp) [reply devel/create_distro_tarball.sh]: git clone [...] src
cd src
git checkout $GIT_TAG

File: devel/create_distro_tarball.sh
Comment (hillsp) [reply devel/create_distro_tarball.sh:144]: Fair enough. No reason we can't change that later

**Comment by hillsp** [comment]
There are MANY things wrong with this script, but it's not significantly worse than the original. Don't forget to update the release docs delete the old script once this lands.

LGTM

**Review by jeffkaufman — APPROVED** [review]
LGTM

(Similar feelings to @hillsp : this script is still a pain, but your changes make it better.)

**Pull request merged**
